client/src/components/register/register.component.js

```diff
@@ -0,0 +1,136 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import Form from 'react-bootstrap/Form';
+import Container from 'react-bootstrap/Container';
+import Button from 'react-bootstrap/Button';
+import Card from 'react-bootstrap/Card';
+import './register.style.css';
+
+export class Register extends Component {
+  constructor(props) {
+    super(props);
+    this.handleForm = this.handleForm.bind(this);
+    this.handleUsername = this.handleUsername.bind(this);
+    this.handleEmail = this.handleEmail.bind(this);
+    this.handlePassword = this.handlePassword.bind(this);
+    this.formIsValid = this.formIsValid.bind(this);
+  }
+  handleForm(e) {
+    e.preventDefault();
+    const credentials = {
+      username: this.refs.inputUsername.value,
+      email: this.refs.inputEmail.value,
+      password: this.refs.inputPassword.value
+    };
+    this.props.handleForm(credentials);
+  }
+
+  handleUsername() {
+    this.props.handleUsername(this.refs.inputUsername);
+  }
+
+  handlePassword() {
```

Comment:
The usage of refs is discouraged in react, since you skip the shadow dom completely and use the regular dom instead. The only cases where refs are a good idea is when:
- Managing focus, text selection, or media playback.
- Integrating with third-party DOM libraries.
- Triggering imperative animations.

See [react docs](https://facebook.github.io/react/docs/refs-and-the-dom.html) and [Robin Wieruch's article](https://www.robinwieruch.de/react-ref-attribute-dom-node/) for more details on this

Since react has what it calls [synthetic events](https://reactjs.org/docs/events.html#form-events) that wrap regular events to work on the shadow dom instead, the best practice is to use them here instead. Thus, a proper way to handle this would be to refactor your handlePassword as such:
```js
handlePassword(e) {
  this.props.handlePassword(e.target.value) // or e.target if you want to pass the element
}
```

Since forms are a pain to make and especially to validate, a lot of people use [Formik](https://reactjs.org/docs/forms.html#fully-fledged-solutions) and it's also suggested by [react docs](https://reactjs.org/docs/forms.html#fully-fledged-solutions)

Reply:
Sure. I can do this.

Reply:
See 706f0ee3bd578a8ebd28f0ef10897541146f9c85